### PR TITLE
Fix its plugin failure on mac

### PR DESCRIPTION
**Pull request by alban-auzeill**
- Fix NoClassDefFoundError related to org.codehaus.groovy.vmplugin.v7.Java7 using java 17 like described by https://github.com/aws/aws-sdk-java-v2/issues/2165
- Fix jdk8 for linux downloaded on a mac

### CODE REVIEW

File: its/plugin/tests/pom.xml [hunk 1]
```diff
@@ -99,105 +102,53 @@
...
+      </plugin>
+    </plugins>
   </build>
 
   <profiles>
     <profile>
-      <id>download-jre-8-linux</id>
-      <activation>
-        <os>
-          <family>unix</family>
```
Comment (alban-auzeill) [inline its/plugin/tests/pom.xml:109]: Unfortunately, `unix` is active on linux and mac. The linux filter should be `<family>unix</family><name>Linux</name>`.

File: its/plugin/tests/pom.xml [hunk 2]
```diff
@@ -99,105 +102,53 @@
...
       </properties>
-      <build>
-        <plugins>
-          <plugin>
-            <groupId>com.googlecode.maven-download-plugin</groupId>
-            <artifactId>download-maven-plugin</artifactId>
-            <executions>
-              <execution>
-                <id>unpack-jre-macosx</id>
-                <phase>package</phase>
```
Comment (alban-auzeill) [inline its/plugin/tests/pom.xml:186]: it's too late to download jdk8 at `<phase>package</phase>` it should be`<phase>generate-test-resources</phase>`

File: its/plugin/tests/pom.xml [hunk 3]
```diff
@@ -99,105 +102,53 @@
...
     <profile>
-      <id>download-jre-8-macosx</id>
+      <id>set-openjdk8-mac</id>
       <activation>
         <os>
           <family>mac</family>
         </os>
       </activation>
       <properties>
-        <jre.dirname>jdk8u256-b01-jre/Contents/Home</jre.dirname>
```
Comment (alban-auzeill) [inline its/plugin/tests/pom.xml:176]: I didn't find any usage of `jre.dirname`

**Review by dorian-burihabwa-sonarsource — APPROVED** [review]
LGTM 👍🏿

**Comment by dorian-burihabwa-sonarsource** [comment]
@alban-auzeill Please remember to verify this

**Pull request merged**